java/springmvc/src/main/java/com/lacunasoftware/pkisuite/controller/AuthenticationCloudExpressController.java

```diff
@@ -0,0 +1,182 @@
+        // signature and the custom state (fileId).
+        TrustServiceSessionResult trssResult = manager.completeAuth(authorizationCode, state);
+		// get the signature session given by the cloud provider
+        String session = trssResult.getSession();
+		// retrieve the nonce stored in the custom state
+		String nonce = trssResult.getCustomState();
+
+		// Get the certificate using the same session
+		CertificateExplorer certificateExplorer = new CertificateExplorer();
+		certificateExplorer.setTrustServiceSession(session);
+		certificateExplorer.setFillContent(true);
+		CertificateExplorerResult certResult = certificateExplorer.open();
+		byte[] certContent = certResult.getCertificate().getContent();
+
+		// Perform a signature using PKIExpress on the nonce field
+		DataSigner dataSigner = new DataSigner();
+		dataSigner.setTrustServiceSession(session);
+		dataSigner.setToSignData(nonce);
+		byte[] signature = dataSigner.sign();		
+
+		// Get an instance of the Authentication class, responsible for contacting PKI Express to
+		// start and complete the authentication operation.
+        Authentication auth = new Authentication();
+		auth.setNonce(nonce);
+		auth.setCertificate(certContent);
+		auth.setSignature(signature);
+
+		// Complete the authentication. Receive as response a AuthCompleteResult instance containing
+		// the following fields:
+		// - The certificate information;
+		// - The validation results;
+		AuthCompleteResult result = auth.complete();
+
+		// Check the authentication result,
+		ValidationResults vr = result.getValidationResults();
+		if (!vr.isValid()) {
+
+			// If the authentication was not successful, we render a page showing that what went
+			// wrong.
+
+			// The toString() method of the ValidationResults object can be used to obtain the
+			// checks performed, but the string contains tabs and new line characters for
+			// formatting, which we'll convert to <br>'s and &nbsp;'s.
+			String vrHtml = vr.toString()
+				.replaceAll("\n", "<br>")
+				.replaceAll("\t", "&nbsp;&nbsp;&nbsp;&nbsp;");
+			model.addAttribute("vrHtml", vrHtml);
+
+			// Render the authentication failed page (templates/authentication-failed.html)
+			return "authentication-express/failed";
+		}
+
+		// At this point, you have assurance that the certificate is valid. Now, you'd typically
+		// query your database for a user that matches one of the certificate's fields, such as
+		// userCert.getEmailAddress() or userCert.getPkiBrazil().getCpf() (the actual field to be
+		// used as key depends on your application's business logic) and set the user as
+		// authenticated with whatever web security framework your application uses. For
+		// demonstration purposes, we'll just render the user's certificate information.
+		model.addAttribute("userCert", result.getCertificate());
+
+		// Render the authentication succeeded page (templates/authentication-success.html)
+		return "authentication-express/success";
```

Comment:
Here, it's returning a view from other example. Shouldn't return something like: `authentication-cloud-express/success`?
I also noticed this view doesn't exist. There is only the `failed` page.

Reply:
Please check https://github.com/LacunaSoftware/PkiSuiteSamples/pull/104/commits/ba06c6680d8c365ab41ef3452979b2df861e5666